doctr/models/detection/linknet/pytorch.py

```diff
@@ -188,18 +193,30 @@ def compute_loss(
 
         seg_target, seg_mask = torch.from_numpy(seg_target).to(dtype=out_map.dtype), torch.from_numpy(seg_mask)
         seg_target, seg_mask = seg_target.to(out_map.device), seg_mask.to(out_map.device)
+        seg_mask = seg_mask.to(dtype=torch.float32)
 
-        # BCE loss
-        bce_loss = F.binary_cross_entropy_with_logits(out_map, seg_target, reduction='none')
+        bce_loss = F.binary_cross_entropy_with_logits(
+            out_map, seg_target, reduction='none',
+            pos_weight=alpha * torch.ones(out_map.shape[1], dtype=torch.float32, device=out_map.device),
+        )
```

Comment:
As discussed offline, my bad, with this design, we only apply alpha to the positive contribution to the loss, while we have to apply (1 -alpha) to the negative part:
```python
bce_loss = bce_loss = F.binary_cross_entropy_with_logits(out_map, seg_target, reduction='none')
```